Update outdated docs (#16407)
kyma-project/kyma: docs/04-operation-guides/troubleshooting/eventing/evnt-04-free-jetstream-storage.md

NATS JetStream backend stopped receiving events due to full storage.

> You observe the following behaviour in the Event Publisher Proxy(EPP):

Reviewer: ```suggestion
You observe the following message in the Event Publisher Proxy (EPP):
```

Author: IMO let's keep it as is, since the one of the points underneath is not referring to message (the HTTP code). what do you think ?

> You observe the following behaviour in the Event Publisher Proxy (EPP):

Reviewer 2: ```suggestion
You observe the following behavior in the Event Publisher Proxy (EPP):
```

> - `507 Insufficient Storage` HTTP Status from EPP on publish request.

Reviewer: ```suggestion
- `507 Insufficient Storage` HTTP Status from Event Publisher Proxy on publish request.
```

Author: Since I already introduce the EPP abbreviation above, I think it's valid to use `EPP` here.

Reviewer: Well, its upto TWs to decide if its okay for public docs. :)

Reviewer 2: Since the abbreviation was introduced before, it's ok to use it later in the doc :)

> - `507 Insufficient Storage` HTTP Status from EPP on publish request.

Reviewer 2: ```suggestion
- `507 Insufficient Storage` HTTP Status from EPP on the publish request.
```

There are several ways to free the space on NATS JetStream backend:

- If the published events are too large, the consumer cannot deliver them fast enough before the storage is full.

>   In that case, either slow down the events publish rate until the events are delivered, or scale the NATS backend with additional replicas.

Reviewer 2: ```suggestion
  In that case, either slow down the events' publish rate until the events are delivered, or scale the NATS backend with additional replicas.
```

- Check the [NATS JetStream backend status](evnt-01-eventing-troubleshooting.md#step-6-check-nats-jetstream-status) and if [the sink is reachable and can accept the events](evnt-01-eventing-troubleshooting.md#step-5-check-if-the-subscription-sink-is-healthy).

- The `Interest` retention policy specifies that events published to the subject are not kept in the stream if they don't match any consumer filter.
  You can delete a Kyma Subscription, which automatically removes all the pending messages in the stream that were published to that Subscription's subject.

> - In case the events publish rate is very high (more than 1.5k events per second), speed up the event dispatching by increasing the `maxInFlightMessages` configuration of the Subscription (default is set to 10) accordingly. Due to low `maxInFlightMessages`, the dispatcher will not be able to keep up with the publisher and as a result the stream size will keep growing. 

Reviewer 2: ```suggestion
- In case the events' publish rate is very high (more than 1.5k events per second), speed up the event dispatching by increasing the `maxInFlightMessages` configuration of the Subscription (default is set to 10) accordingly. Due to low `maxInFlightMessages`, the dispatcher will not be able to keep up with the publisher, and as a result, the stream size will keep growing. 
```

There are several ways to free the space on NATS JetStream backend:

- If the published events are too large, the consumer cannot deliver them fast enough before the storage is full.

>   In that case, either slow down the events pace until the events are delivered, or scale the NATS backend with additional replicas.

Reviewer: ```suggestion
  In that case, either slow down the events publish rate until the events are delivered, or scale the NATS backend with additional replicas.
```

- Check the [NATS JetStream backend status](evnt-01-eventing-troubleshooting.md#step-6-check-nats-jetstream-status) and if [the sink is reachable and can accept the events](evnt-01-eventing-troubleshooting.md#step-5-check-if-the-subscription-sink-is-healthy).

- The `Interest` retention policy specifies that events published to the subject are not kept in the stream if they don't match any consumer filter.
  You can delete a Kyma Subscription, which automatically removes all the pending messages in the stream that were published to that Subscription's subject.

> - In case the events-per-second rate is very high (more than 1.5k per second) and the `maxInFlightMessages` property of the NATS JetStream backend is low (10 per default)

Reviewer: ```suggestion
- In case the events publish rate is very high (more than 1.5k events per second), speed up the event dispatching by increasing the `maxInFlightMessages` configuration of the Subscription (default is set to 10) accordingly. Due to low `maxInFlightMessages`, the dispatcher will not be able to keep up with the publisher and as a result the stream size will keep growing. 
```

> the dispatcher will not be able to keep up with the publisher and the stream size will keep growing. To speed to the event dispatching, increase the `maxInFlightMessages` property in the Eventing Controller. 

Reviewer: ```suggestion

```
